edgecompute/examples/traffic-routing/url-shortner/main.js

```diff
@@ -0,0 +1,147 @@
+  var name = params.get('name');
+  var url = params.get('url');
+  var status; 
+  var body;  
+  var checkID;
+  
+  // Supported operations are 'add', 'update' and 'delete'
+  switch (params.get('op')) {
+    case 'add':
+      // If a custom redirect name was not defined, then randomly create a name to use as the key
+      if (name === undefined || name === null) {
+        name = Math.random().toString(36).substr(2, 10);
+      }
+
+      // Add redirect to EKV
+      try {
+        checkID = await edgeKv.getText({ item: name, default_value: undefined });
+
+        if(checkID != undefined) {
+          status = "Error - Redirect already exists for /r/" + name;  
+          body = '<b>' + status + '<b><br>';
+          logger.log(status);
+        } else {
+          await edgeKv.putText({ item: name, value: url });
+          status = "Redirect Successfully added";
+          body = '<b>' + status + '<b><br> \
+            <p>URL: <a href="https://' +request.host + '/r/' + name + ' ">https://' +request.host + '/r/' + name +'</a></p>\
+            <p>Redirects to: <a href=" ' + url + ' ">' + url +'</a></p>'; 
+        }
+      } catch (error) {        
+        status = "error creating redirect";        
+        body = '<b>' + status + '<b><br>';
+        logger.log(status);
+      }
+  
+      return Promise.resolve(createResponse(200, {'Content-Type': ['text/html'] }, '<html><body>' + body + '</body></html>'));
+      break;
+      
+    case 'update':
+      if (name === undefined || url === undefined || name === null || url === null) {
+        status = "error - missing name or url";
+      }
+
+      try {
+        // Check to see if the redirect exists before we attempt to delete it
+        checkID = await edgeKv.getText({ item: name, default_value: undefined });
+        if(checkID != undefined) {
+          // Update redirect
+       		await edgeKv.putText({ item: name, value: url });
+          status = "Redirect Successfully Updated"
+          body = '<b>' + status + '<b><br>\
+          <p>URL: <a href="https://' +request.host + '/r/' + name + ' ">https://' +request.host + '/r/' + name +'</a></p>\
+          <p>Redirects to: <a href=" ' + url + ' ">' + url +'</a></p>';
+          
+        } else {
+          // Redirect doesn't exist
+          status = "Error - Redirect doesn't exist for /r/" + name; 
+          body = '<b>' + status + '<b><br>';
+          logger.log(status); 
+        }
+      } catch (error) {        
+        status = "Error updating redirect: " + error;
+        body = '<b>' + status + '<b><br>';
+        logger.log(status); 
+      }
+
+      return Promise.resolve(createResponse(200, {'Content-Type': ['text/html'] }, '<html><body>' + body + '</body></html>'));			
+      break;
+      
+    case 'delete':
+      if (name === undefined || name === null) {
+        status = "error - missing name";
+        body = '<b>' + status + '<b><br>';
+      }
+
+      try {

```

Comment:
There is absolutely no need to check existence of the item before deleting it, calling delete will delete if it exists or be a no-op essentially. In fact it is even worse to check since in cases when the item was just recently created, the check may not retrun anything (due to eventual consistency property of EdgeKV) and then the delete would not be called. Whereas calling delete without any checks is guaranteed to delete an item if it exists regardless.